File: Serene/Serene.Script/Imports/ServerImports/BasicSamples.FilteredLookupInDetailForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.BasicSamples
{
    public partial class FilteredLookupInDetailForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "BasicSamples.FilteredLookupInDetail";

        public FilteredLookupInDetailForm(string idPrefix) : base(idPrefix) {}

        public Northwind.CustomerEditor CustomerID { [InlineCode("{this}.w('CustomerID', Serene.Northwind.CustomerEditor)")] get; private set; }
        public DateEditor OrderDate { [InlineCode("{this}.w('OrderDate', Serenity.DateEditor)")] get; private set; }
        public LookupEditor CategoryID { [InlineCode("{this}.w('CategoryID', Serenity.LookupEditor)")] get; private set; }
        public FilteredLookupDetailEditor DetailList { [InlineCode("{this}.w('DetailList', Serene.BasicSamples.FilteredLookupDetailEditor)")] get; private set; }
    }
}

File: Serene/Serene.Core/Imports/ServerTypings/BasicSamples.FilteredLookupInDetailForm.ts
namespace Serene.BasicSamples {
    export interface FilteredLookupInDetailForm {
        CustomerID: Northwind.CustomerEditor;
        OrderDate: Serenity.DateEditor;
        CategoryID: Serenity.LookupEditor;
        DetailList: FilteredLookupDetailEditor;
    }

    export class FilteredLookupInDetailForm extends Serenity.PrefixedContext {
        static formKey = 'BasicSamples.FilteredLookupInDetail';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!FilteredLookupInDetailForm.init)  {
                FilteredLookupInDetailForm.init = true;

                var s = Serenity;
                var w0 = Northwind.CustomerEditor;
                var w1 = s.DateEditor;
                var w2 = s.LookupEditor;
                var w3 = FilteredLookupDetailEditor;

                Q.initFormType(FilteredLookupInDetailForm, [
                    'CustomerID', w0,
                    'OrderDate', w1,
                    'CategoryID', w2,
                    'DetailList', w3
                ]);
            }
        }
    }
}
